stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java

```diff
@@ -16,29 +14,26 @@ class UserRepositoryTest {
     @BeforeEach
     void setUp() {
         userRepository = new UserRepository();
-        UserRepository.userMap.put("1", new User("1", "user1", "phone1"));
-        UserRepository.userMap.put("2", new User("2", "user2", "phone2"));
-        UserRepository.userMap.put("3", new User("3", "user3", "phone3"));
-        UserRepository.userMap.put("4", new User("4", "user4", "phone4"));
-    }
-
-    @AfterEach
-    void tearDown() {
-        UserRepository.userMap = new HashMap<>();
+        userRepository.save(new User("1", "user1", "phone1"));
+        userRepository.save(new User("2", "user2", "phone2"));
+        userRepository.save(new User("3", "user3", "phone3"));
+        userRepository.save(new User("4", "user4", "phone4"));
     }
 
     @Test
     void createUser() {
+        int expectedCount = userRepository.count() + 1;
         userRepository.save(new User("5", "user5", "phone5"));
-        assertEquals(5, UserRepository.userMap.size());
-        assertTrue(UserRepository.userMap.containsKey("5"));
+        assertEquals(expectedCount, userRepository.count());
```

Comment:
This is a strange test in my opinion. If you are testing a method, why are you testing it with another method that has not yet been tested? Whereas your method returns a value.
Maybe you shouldn't make the repository private?